packages/eslint-plugin/src/rules/shorthand-property-sorting/index.ts

```diff
@@ -0,0 +1,140 @@
+import { shorthandBuckets, kebabCase, type ShorthandProperties } from '@compiled/utils';
+import type { Rule, Scope } from 'eslint';
+
+import { isStyled, isCss, isCssMap } from '../../index';
+
+const fixProperties = (context: Rule.RuleContext, node: Rule.Node) => {
+  context.report({
+    node: node,
+    messageId: 'shorthand-first',
+    fix: (fixer) => {
+      if (node.type === 'ObjectExpression' && node.properties.length > 0) {
+        // sort the properties by depth
+        const sortedProperties = node.properties.slice().sort((a, b) => {
+          if (
+            a.type === 'Property' &&
+            a.key.type === 'Identifier' &&
+            b.type === 'Property' &&
+            b.key.type === 'Identifier'
+          ) {
+            const propA = kebabCase(a.key.name) as ShorthandProperties;
+            const propB = kebabCase(b.key.name) as ShorthandProperties;
+
+            return shorthandBuckets[propA] - shorthandBuckets[propB];
+          }
+          return 0;
+        });
+
+        const sourceCode = context.getSourceCode();
+        const sortedCode = sortedProperties
+          .map((property) => sourceCode.getText(property))
+          .join(', ');
+
+        // Replace the old object expression with the new sorted one
+        const newObjectExpression = `{ ${sortedCode} }`;
+
+        return fixer.replaceText(node, newObjectExpression);
+      }
+      return null;
+    },
+  });
+};
+
+const callExpressionCorrectImport = (node: Rule.Node, references: Scope.Reference[]): boolean => {
+  return (
+    node.type === 'CallExpression' &&
+    (isCss(node.callee as Rule.Node, references) ||
+      isCssMap(node.callee as Rule.Node, references) ||
+      isStyled(node.callee as Rule.Node, references))
+  );
+};
+
+export const shorthandFirst: Rule.RuleModule = {
+  meta: {
+    docs: {
+      recommended: true,
+      description:
+        "At build time, Compiled automatically sorts shorthand properties (like `font` and `border`) so that they come before any longhand properties (like `fontSize` and `borderTopColor`) defined on the component. This means that longhand properties will always override shorthand properties. This rule enforces that the order in which the properties appear in a component's source code matches the actual ordering the properties will have at build time and runtime.",
```

Comment:
I may be missing something, but I think this rule ensures that redundant CSS won't cause unwanted side effects

Reply:
As discussed internally on a Confluence page, I think we should change the wording in both rule description and README. 

Below my understanding; up to you on how to re-write it 

Having a longhand before a shorthand is not valid CSS. While this is usually not a problem, it becomes a problem in Compiled where classes are generated for each CSS declaration and rules are then sorted at build time during stylesheet extraction. As a result, this could cause unwanted side effects when CSS longhands are redundant.